autogen.sh

```diff
@@ -1,3 +1,11 @@
 #!/usr/bin/env sh
+
+if [[ $TRAVIS_OS_NAME == osx ]]; 
+	then 
+		monoVer=$(mono --version | head -n 1 | cut -d' ' -f 5)
+		prefix="/Library/Frameworks/Mono.framework/Versions/$monoVer"; 
+	else prefix="/usr"; 
```

Comment:
It would, yes.  If that's not preferred I can wrap the call to autogen.sh in another script that travis would be updated to use, then revert autogen.sh back to its original form.

So I'd add a `travis-autogen.sh` that contained the OS-prefix logic, and then called autogen.sh with the specified prefix, then change `travis.tml` to call `travis-autogen.sh` instead of `autogen.sh`.

Thoughts?